src/Compilers/CSharp/Test/Emit3/FieldKeywordTests.cs

```diff
@@ -1650,150 +1650,1327 @@ .maxstack  1
+                    IL_000d:  ret
+                }
+                """);
+            verifier.VerifyIL("C9..ctor(int)", $$"""
+                {
+                    // Code size       14 (0xe)
+                    .maxstack  2
+                    IL_0000:  ldarg.0
+                    IL_0001:  call       "object..ctor()"
+                    IL_0006:  ldarg.0
+                    IL_0007:  ldarg.1
+                    IL_0008:  callvirt   "void C9.P9.{{setter}}"
+                    IL_000d:  ret
+                }
+                """);
+            verifier.VerifyIL("C3..ctor(int, out int)", $$"""
+                {
+                    // Code size       22 (0x16)
+                    .maxstack  2
+                    IL_0000:  ldarg.0
+                    IL_0001:  call       "object..ctor()"
+                    IL_0006:  ldarg.2
+                    IL_0007:  ldarg.0
+                    IL_0008:  callvirt   "int C3.P3.get"
+                    IL_000d:  stind.i4
+                    IL_000e:  ldarg.0
+                    IL_000f:  ldarg.1
+                    IL_0010:  stfld      "int C3.F3"
+                    IL_0015:  ret
+                }
+                """);
+            verifier.VerifyIL("C6..ctor(int, out int)", $$"""
+                {
+                    // Code size       22 (0x16)
+                    .maxstack  2
+                    IL_0000:  ldarg.0
+                    IL_0001:  call       "object..ctor()"
+                    IL_0006:  ldarg.2
+                    IL_0007:  ldarg.0
+                    IL_0008:  callvirt   "int C6.P6.get"
+                    IL_000d:  stind.i4
+                    IL_000e:  ldarg.0
+                    IL_000f:  ldarg.1
+                    IL_0010:  stfld      "int C6.F6"
+                    IL_0015:  ret
+                }
+                """);
+            verifier.VerifyIL("C9..ctor(int, out int)", $$"""
+                {
+                    // Code size       22 (0x16)
+                    .maxstack  2
+                    IL_0000:  ldarg.0
+                    IL_0001:  call       "object..ctor()"
+                    IL_0006:  ldarg.2
+                    IL_0007:  ldarg.0
+                    IL_0008:  callvirt   "int C9.P9.get"
+                    IL_000d:  stind.i4
+                    IL_000e:  ldarg.0
+                    IL_000f:  ldarg.1
+                    IL_0010:  stfld      "int C9.F9"
+                    IL_0015:  ret
+                }
+                """);
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void ConstructorAssignment_02C(bool useInit)
+        {
+            string setter = useInit ? "init" : "set";
+            string source = $$"""
+                #pragma warning disable 649
+                using System;
+                struct C1
+                {
+                    public int F1;
+                    public int P1 { get; }
+                    public C1(int i) { P1 += i; }
```

Comment:
It seems like this and the other scenarios in this test would be more illustrative if `F1` were explicitly assigned after the `P1 += i;` statement. That would show whether the property access is causing auto-default of the whole instance.

Same for the `++`, `--`, `??=` tests below.

Reply:
Thanks, that makes sense.